packages/graph-explorer/src/hooks/useExpandNode.tsx

```diff
@@ -0,0 +1,281 @@
+import {
+  PropsWithChildren,
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import { useNotification } from "../components/NotificationProvider";
+import type {
+  NeighborsRequest,
+  NeighborsResponse,
+} from "../connector/useGEFetchTypes";
+import { explorerSelector } from "../core/connector";
+import useEntities from "./useEntities";
+import { useRecoilValue } from "recoil";
+import { UseQueryResult, useQueries, useQuery } from "@tanstack/react-query";
+import { neighborsCountQuery, neighborsQuery } from "../connector/queries";
+import useDisplayNames from "./useDisplayNames";
+import { Vertex } from "../@types/entities";
+import { isEqual } from "lodash";
+
+/*
+
+# Developer Note
+
+Since the expand request could come from the GraphViewer or the sidebar, we must
+encapsulate the logic within a React context. This ensures the dependent queries
+will be run regardless if the view is unmounted.
+
+*/
+
+type ExpandNodeFilters = Omit<
+  NeighborsRequest,
+  "vertexId" | "idType" | "vertexType"
+>;
+
+export type ExpandNodeRequest = {
+  vertex: Vertex;
+  filters?: ExpandNodeFilters;
+};
+
+export type ExpandNodeContextType = {
+  expandNode: (vertex: Vertex, filters?: ExpandNodeFilters) => void;
+  reset: () => void;
+  query: UseQueryResult<NeighborsResponse | null, Error>;
+};
+
+const ExpandNodeContext = createContext<ExpandNodeContextType | null>(null);
+
+export function ExpandNodeProvider(props: PropsWithChildren) {
+  // Wires up node count query in response to new nodes in the graph
+  useUpdateNodeCounts();
+
+  const [expandNodeRequest, setExpandNodeRequest] =
+    useState<ExpandNodeRequest | null>(null);
+
+  // Perform the query when a request exists
+  const request: NeighborsRequest | null = expandNodeRequest && {
+    vertexId: expandNodeRequest.vertex.data.id,
+    idType: expandNodeRequest.vertex.data.idType,
+    vertexType:
+      expandNodeRequest.vertex.data.types?.join("::") ??
+      expandNodeRequest.vertex.data.type,
+    ...expandNodeRequest.filters,
+  };
+  const explorer = useRecoilValue(explorerSelector);
+  const query = useQuery(neighborsQuery(request, explorer));
```

Comment:
I'm going to try and use `useMutation()` here instead since it is a better fit for our use case. While we are technically querying data, we are using that data to update local state of added nodes & edges in the graph, which is a mutation of a kind.

Reply:
The mutation worked way better than I expected. The change has been pushed up.